guvnor-ala/guvnor-ala-spi/src/main/java/org/guvnor/ala/marshalling/MarshallerRegistry.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.marshalling;
+
+/**
+ * General purpose contract for the registering of marshallers.
+ */
+public interface MarshallerRegistry {
+
+    void register(final Class clazz,
```

Comment:
I tried it, and it's a good idea from the static typing perspective.
But It was not possible to make  work altogether  when the marshallers needs to be injected by using CDI in the code below.

@ApplicationScoped
public class VFSMarshallerRegistry
        extends BaseMarshallerRegistry {

    private static final Logger logger = LoggerFactory.getLogger(VFSMarshallerRegistry.class);

    private Instance<Marshaller> marshallerInstance;

    public VFSMarshallerRegistry() {
        //Empty constructor for Weld proxying
    }

    @Inject
    public VFSMarshallerRegistry(final @Any Instance<Marshaller> marshallerInstance) {
        this.marshallerInstance = marshallerInstance;
    }

    @PostConstruct
    public void init() {
        marshallerInstance.iterator().forEachRemaining(marshaller -> {
            logger.debug("Registering marshaller into VFSMarshallerRegistry, marshaller: " + marshaller);
            register(marshaller.getType(),
                     marshaller);
        });
    }
}